### ISSUE TITLE
Hot module replace dont work in this simple project

### ISSUE THREAD

**Issue by prsolucoes** [webpack/webpack#3271]
**I'm submitting a bug report**

**Video showing the problem**
https://youtu.be/mQ0-C3uJpfg

**Webpack version:**
2.0.15 beta

**Please tell us about your environment:**
OSX 10.12.1

**Current behavior:**
Webpack shows on console that it is reloading, but it dont replace the new content on the HTML.

**Expected/desired behavior:**
On change HTML file, the HMR need change the HTML on browser

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem along with a gist/jsbin of your webpack configuration.** 

Copy and paste this lines on terminal/console:

> git clone https://github.com/prsolucoes/angular2-starter-project
> cd angular2-starter-project
> npm install
> npm start

Open "http://localhost:9000" on your browser and after open change any HTML inside "src/app/components".

* **What is the expected behavior?**
The new HTML content need be displayed in the browser

* **Browser:** all
* **Language:** TypeScript 2

**Comment by nullptr128** [comment]
I've the same issue as well in my project. It does not output "compiling" log multiple times, but the rest is exacly the same, when I load webpack-dev-server page:

[HMR] Waiting for update signal from WDS...
[WDS] Hot Module Replacement enabled.

When I save file, this appears:

[WDS] App updated. Recompiling...
[WDS] App hot update...

But nothing happens in page, no modules are hot-swapped, if I edit css file then changes are not reflected to page. Of course when I F5, the changes apply.

What I have noticed is that brower does not download new parts. That how it should look like:

https://camo.githubusercontent.com/0fb51bd52bb6109147eccefb92a93fea0acb4934/687474703a2f2f692e696d6775722e636f6d2f6f4f6330696b562e706e67

But on my browser, no new requests are made to server, it just stay calm.

**Comment by TheLarkInn** [comment]
@gdi2290 do you think you could assist with this? My ng hmr experience is somewhat limited.

**Comment by sortofsleepy** [comment]
Any updates on this? I'm also having trouble getting HMR to work under Webpack 2 though I've updated to the release candidate(2.2.0-rc.0) 

Funny thing is, the re-build does appear to be working from what I can see. In the warning about entry point size limits where it lists all the files, I see my bundle file as well as another filename `<long hash>.hot-update.js` and if I `view-source` and check the javascript, the bundle appears to have updated as well.

Now I have no idea about the inner workings, but from what I can see, the issue, for me at least, appears to be that the updated code doesn't get appended to the DOM for some reason.

I see theres an example on the new Webpack site for how to get setup with React, any chance someone's found/written a more barebones example out there without a framework just so I can double check my config? I've basically taken that example and removed the React bits with no luck.

**Comment by aunz** [comment]
Hello, @prsolucoes could you post all your webpack configs, it will be easier to find bugs for you.

Anyway, from your video, I assume you are also using webpack-dev-server. If you look into the directory `node_modules\webpack-dev-server\package.json`, what is the version?

If it's 1.x, then you should reinstall it with `npm i -D webpack-dev-server@2.2.0-rc.0` 

If you want to know more about the inner workings, have a look at this line in [v1.16](https://github.com/webpack/webpack-dev-server/blob/v1.16.2/client/index.js#L123) vs this line in [v.2.2](https://github.com/webpack/webpack-dev-server/blob/v2.2.0-rc.0/client/index.js#L145).

**Comment by namxam** [comment]
I just updated to 2.2 after using the betas and release candidates. And for some reason, it is now showing the same behaviour as @sortofsleepy is having. I see webpack is building the *.hot-update.js files. But somehow the frontend is not picking them up and only shows me 

```
[HMR] Cannot find update. Need to do a full reload!
[HMR] (Probably because of restarting the webpack-dev-server)
```

**Comment by sortofsleepy** [comment]
Gagh! haha totally forgot I posted here. 😛  

@namxam - I actually have it working now with `2.2.0` (installed after seeing [this Medium post](https://medium.com/webpack/webpack-2-2-the-final-release-76c3d43bf144#.4g81izh8x) )for both webpack and the dev server. 

This is what my typical config looks like that worked last I checked if you wanna try it out again.(Adjust things as necessary) 
https://gist.github.com/sortofsleepy/cc18e2c013d4623cd3169b01bb1ede9e
